acme/dns_challenge.go

```diff
@@ -144,7 +145,8 @@ func checkAuthoritativeNss(fqdn, value string, nameservers []string) (bool, erro
 	return true, nil
 }
 
-// dnsQuery sends a DNS query to the given nameserver.
+// dnsQuery will attempt 3 times, a query to the given nameserver.
+// If the given nameserver is `RecursiveNameserver`, it will attempt the `altRecursiveNameserver` on the 2nd and 3rd attempts.
 // The nameserver should include a port, to facilitate testing where we talk to a mock dns server.
 func dnsQuery(fqdn string, rtype uint16, nameserver string, recursive bool) (in *dns.Msg, err error) {
```

Comment:
Just an idea: But since we now use multiple recursive server, what do you think about making dnsQuery() and FindZoneByFqdn() accept a slice of nameservers and declare the default servers like this:

```Golang
var RecursiveNameservers = []string{
	"google-public-dns-a.google.com:53",
	"google-public-dns-b.google.com:53",
}
```

This would make it also quite simple to actually alternate between servers on each retry:

```Golang
	// Try up to 3 times (returning early if nil errors)
	for i := 1; i <= 3; i++ {
		ns := nameservers[i % len(nameservers)]
		in, err = dns.Exchange(m, ns)
```